Sources/Action.swift

```diff
@@ -10,15 +10,91 @@ import Result
 /// Actions enforce serial execution. Any attempt to execute an action multiple
 /// times concurrently will return an error.
 public final class Action<Input, Output, Error: Swift.Error> {
+	public enum ExecutionStrategy {
+		/// The default execution strategy. Only one worker is allowed to be
+		/// executing at a time. Any further start attempt during the execution
+		/// is rejected with `ActionError.disabled`.
+		case `default`
+
+		/// The concurrent execution strategy. Up to `limit` workers are allowed
+		/// to be executing at a given time. Any further start attempt during
+		/// the execution would:
+		///
+		/// 1. succeed immediately if the concurrent limit has not been reached;
+		///    or
+		/// 2. be queued up to the given capacity if the concurrent limit has
+		///    been reached; or
+		/// 3. be rejected with `ActionError.disabled` if the capacity limit has
+		///    been reached.
+		///
+		/// - parameters:
+		///   - limit: The concurrent limit of the `Action`.
+		///   - capacity: The capacity limit of the `Action`.
+		///
+		/// - note: `capacity` is inclusive of `limit`.
+		case concurrent(limit: UInt, capacity: UInt)
```

Comment:
I'm not sure that opening it up to concurrency is a good idea. That adds quite a bit of complexity.

Maybe we should limit it to exclusive strategies to start?